TA_dataset/bin/dataset_api.py

```diff
@@ -82,34 +103,42 @@ def ds_lrq_facet_values(
             filter_=filter, name=name, max_values=max_values
         ),
     )
-    return ds_lrq_run_loop(client=client, body=body)
+    return ds_lrq_run_loop(client=client, body=body, logger=logger)
 
 
 # Executes LRQ run loop of launch-ping-remove API requests until the query completes
 # with a result
+# Returns tuple - value, error message
 def ds_lrq_run_loop(
-    client: AuthenticatedClient, body: PostQueriesLaunchQueryRequestBody
+    client: AuthenticatedClient, body: PostQueriesLaunchQueryRequestBody, logger: Logger
 ):
     body.query_priority = PostQueriesLaunchQueryRequestBodyQueryPriority.HIGH
     response = post_queries.sync_detailed(client=client, json_body=body)
+    logger.info(response)
     result = response.parsed
-    forward_tag = response.headers["x-dataset-query-forward-tag"]
-    steps_done = result.steps_completed
-    steps_total = result.steps_total
-    query_id = result.id
-    while steps_done < steps_total:
-        response = get_queries.sync_detailed(
-            id=query_id,
-            query_type=body.query_type,
-            client=client,
-            last_step_seen=steps_done,
-            forward_tag=forward_tag,
-        )
-        result = response.parsed
+    logger.info(result)
+    logger.info(response.content)
+    if result:
+        forward_tag = response.headers["x-dataset-query-forward-tag"]
         steps_done = result.steps_completed
-    delete_queries.sync_detailed(id=query_id, client=client, forward_tag=forward_tag)
+        steps_total = result.steps_total
+        query_id = result.id
+        while steps_done < steps_total:
+            response = get_queries.sync_detailed(
+                id=query_id,
+                query_type=body.query_type,
+                client=client,
+                last_step_seen=steps_done,
+                forward_tag=forward_tag,
+            )
+            result = response.parsed
+            steps_done = result.steps_completed
+        delete_queries.sync_detailed(
+            id=query_id, client=client, forward_tag=forward_tag
+        )
 
-    return result
+        return result, None
+    return None, json.loads(response.content)
```

Comment:
So we return the whole JSON response body as an error? Do we want to return just a specific field or similar?

Reply:
The error is passed to the function - `search_error_exit` - https://github.com/scalyr/dataset-addon-for-splunk/blob/main/TA_dataset/bin/dataset_search_command.py#L119 - which expects dictionary, where there is attribute `message` - and this is the same in both cases.